keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/pause/ResourcePauser.kt

```diff
@@ -67,6 +68,11 @@ class ResourcePauser(
   fun resumeApplication(application: String) {
     log.info("Resuming application $application")
     pausedRepository.resumeApplication(application)
+    resourceRepository.getResourcesByApplication(application)
+      .forEach {
+        // helps a user not be confused by an out of date status from before a pause
+        publisher.publishEvent(ResourceActuationResumed(it))
```

Comment:
If I:
1. pause a resource (to try to fix something), then
2. paused the entire application (because oh my), then
3. unpause the application (because I fixed my things)

I don't fully understand how event publishing works in keel, but this won't actually remove the pause from (1), will it? Because that's an override that's always derived via the PausedRepository?

Reply:
It won't remove the pause from the individual resource, you're correct. The status will still show as paused, and the most recent event in the history will also be paused. The second event will be a resumed event which will be irrelevant. It would be clearer to have a message on each event saying at what level it was resumed.

As I've been working with pausing and status and these changes I have realized it's somewhat complicated. I think I'll take another look at it next week and think about how I can simplify or make things easier to work with.